In [1]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.neighbors import KNeighborsClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split, GridSearchCV,RandomizedSearchCV
from sklearn.metrics import recall_score,precision_score,f1_score,accuracy_score, confusion_matrix, classification_report
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
pd.set_option("display.max_columns", None)


In [2]:
lr = pd.read_excel('/Users/chirag/Desktop/Office/HZ/YTD Lead Report.xlsx')
pr = pd.read_csv('/Users/chirag/Desktop/Office/HZ/YTD Payment Report.csv')


In [3]:
pr

Product Type Quote Creation Date           Source Partner name  \
0          Four Wheeler          03/01/2022  Assisted module    FGNG POSP   
1    Two Wheeler Retail          04/01/2022  Assisted module    FGNG POSP   
2    Two Wheeler Retail          04/01/2022  Assisted module    FGNG POSP   
3    Two Wheeler Retail          05/01/2022  Assisted module      Biznext   
4    Two Wheeler Retail          06/01/2022  Assisted module    FGNG POSP   
..                  ...                 ...              ...          ...   
687  Two Wheeler Retail          20/03/2022  Assisted module      Biznext   
688  Two Wheeler Retail          20/03/2022  Assisted module   B2B2C_PAY1   
689  Two Wheeler Retail          20/03/2022  Assisted module   B2B2C_PAY1   
690  Two Wheeler Retail          20/03/2022  Assisted module   B2B2C_PAY1   
691  Two Wheeler Retail          20/03/2022  Assisted module   B2B2C_PAY1   

     Branch name        Agent name  Reporting Manager  Quote Transaction ID  \
0            NaN  Ashwini Deshmukh                NaN               3974076   
1            NaN  Ashwini Deshmukh                NaN               3974893   
2            NaN  Ashwini Deshmukh                NaN               3975120   
3            NaN       Vinay Kedia                NaN               3976502   
4            NaN  Ashwini Deshmukh                NaN               3977578   
..           ...               ...                ...                   ...   
687          NaN       Rhaul Ahmed                NaN               4133386   
688          NaN               NaN                NaN               4133407   
689          NaN               NaN                NaN               4133428   
690          NaN               NaN                NaN               4133432   
691          NaN               NaN                NaN               4133490   

    Vehicle Registration No     Mobile No                    Email ID  \
0                 SK01P5828  7.407383e+09          drajrai1@gmail.com   
1                MH12MJ4792  9.545451e+09         pranjali3@gmail.com   
2                 AP13Q8370  9.866762e+09         raounihyd@gmail.com   
3                MH43BH3757  9.167307e+09    saurabhshanu00@gmail.com   
4                MH05DV2349  8.983738e+09    noorshaikh1247@gmail.com   
..                      ...           ...                         ...   
687               TR05C7777  6.361330e+09     rhaulahmed999@gmail.com   
688              MH02CN8255  8.369285e+09           Ave3055@gmail.com   
689               AS16B5182  8.011946e+09   mojamelhoque019@gmail.com   
690               AS17H5326  8.473001e+09  enamulhussain797@gmail.com   
691              AS01EK1009  8.876389e+09       shineghy001@gmail.com   

    Vehicle API Vehicle Owner name Vehicle Ownership Type  \
0           Yes             DHIRAJ             Individual   
1           Yes           PRANJALI             Individual   
2           Yes             M APPA             Individual   
3           Yes            SAURABH             Individual   
4           Yes              SUMIT             Individual   
..          ...                ...                    ...   
687         Yes     MD RHAUL AHMED             Individual   
688         Yes     MILAN A MISHRA             Individual   
689         Yes      MD SATTAR ALI             Individual   
690         Yes       ENAMUL HOQUE             Individual   
691         Yes         AJIT GOGOI             Individual   

                                   Vehicle Make  \
0                       HYUNDAI MOTOR INDIA LTD   
1                    INDIA YAMAHA MOTOR PVT LTD   
2    HONDA MOTORCYCLE AND SCOOTER INDIA (P) LTD   
3                                BAJAJ AUTO LTD   
4               SUZUKI MOTORCYCLE INDIA PVT LTD   
..                                          ...   
687                                        HERO   
688                                       HONDA   
689                                       BAJAJ   
690              

In [4]:
l=lr[['Product Type','Partner name','Vehicle Make','Manufacturing Date','Previous Year Claim','Total Premium with GST','State']]

f=pr[['Product Type','Partner name','Vehicle Make','Manufacturing Date','Previous Year Claim','Total Premium with GST','State']]


In [5]:
l.dropna(inplace=True)

/opt/anaconda3/lib/python3.9/site-packages/pandas/util/_decorators.py:311: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return func(*args, **kwargs)


In [6]:
l['Sucess status'] = l['Sucess status']=0

/var/folders/pf/7k3lhqkn50g7d2nc7xq272940000gp/T/ipykernel_40936/2274917596.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  l['Sucess status'] = l['Sucess status']=0


In [7]:
f['Sucess status'] = f['Sucess status']=1

/var/folders/pf/7k3lhqkn50g7d2nc7xq272940000gp/T/ipykernel_40936/2828349794.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  f['Sucess status'] = f['Sucess status']=1


In [8]:
df=pd.concat([l,f])

In [9]:
df

Product Type Partner name Vehicle Make  Manufacturing Date  \
0    Two Wheeler Retail      Biznext   BAJAJ AUTO              2017.0   
4    Two Wheeler Retail    FGNG POSP   BAJAJ AUTO              2012.0   
6    Two Wheeler Retail    FGNG POSP        HONDA              2013.0   
7    Two Wheeler Retail    FGNG POSP   BAJAJ AUTO              2018.0   
19   Two Wheeler Retail    FGNG POSP        HONDA              2018.0   
..                  ...          ...          ...                 ...   
687  Two Wheeler Retail      Biznext         HERO              2014.0   
688  Two Wheeler Retail   B2B2C_PAY1        HONDA              2012.0   
689  Two Wheeler Retail   B2B2C_PAY1        BAJAJ              2011.0   
690  Two Wheeler Retail   B2B2C_PAY1         HERO              2018.0   
691  Two Wheeler Retail   B2B2C_PAY1        HONDA              2020.0   

    Previous Year Claim  Total Premium with GST        State  Sucess status  
0                    No                  887.36  Maharashtra              0  
4                    No                 1276.76    Rajasthan              0  
6                    No                 1176.46       Orissa              0  
7                    No                 2293.92  Maharashtra              0  
19                   No                  887.36       Orissa              0  
..                  ...                     ...          ...            ...  
687                  No                 1401.84      Tripura              1  
688                  No                  887.36  Maharashtra              1  
689                  No                 1797.14        Assam              1  
690                  No                 1667.34        Assam              1  
691                  No                 1477.36        Assam              1  

[5217 rows x 8 columns]

In [10]:
df = df.sample(frac=1).reset_index(drop=True)

In [11]:
df.dtypes

Product Type               object
Partner name               object
Vehicle Make               object
Manufacturing Date        float64
Previous Year Claim        object
Total Premium with GST    float64
State                      object
Sucess status               int64
dtype: object

In [12]:
df['Product Type'] = df['Product Type'].astype('category')
df['Product Type'] = df['Product Type'].cat.codes
df['Partner name'] = df['Partner name'].astype('category')
df['Partner name'] = df['Partner name'].cat.codes
df['Vehicle Make'] = df['Vehicle Make'].astype('category')
df['Vehicle Make'] = df['Vehicle Make'].cat.codes
df['Previous Year Claim'] = df['Previous Year Claim'].astype('category')
df['Previous Year Claim'] = df['Previous Year Claim'].cat.codes
df['State'] = df['State'].astype('category')
df['State'] = df['State'].cat.codes

In [13]:
df

Product Type  Partner name  Vehicle Make  Manufacturing Date  \
0                1             2            18              2010.0   
1                0             2            46              2021.0   
2                1             2            13              2017.0   
3                1             2             3              2010.0   
4                1             0            18              2014.0   
...            ...           ...           ...                 ...   
5212             1             0            58              2018.0   
5213             0             2            10              2013.0   
5214             1             2            48              2018.0   
5215             1             2             3              2000.0   
5216             1             2            18              2013.0   

      Previous Year Claim  Total Premium with GST  State  Sucess status  
0                       0                  887.36     55              0  
1                       1                 9523.78     33              0  
2                       0                 1116.28     66              0  
3                       0                 1325.14     19              0  
4                       0                  887.36     50              0  
...                   ...                     ...    ...            ...  
5212                    0                 1283.84     66              1  
5213                    0                 6228.04     22              0  
5214                    0                 1797.14     47              0  
5215                    0                 1410.10     29              0  
5216                    0                 1371.16     59              0  

[5217 rows x 8 columns]

In [14]:
df['Sucess status'].value_counts()

0    4525
1     692
Name: Sucess status, dtype: int64

In [15]:
X = df.iloc[:,:6]
y = df['Sucess status']

In [16]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=0.3, random_state=0)

In [17]:
from sklearn.linear_model import LogisticRegression
classification = LogisticRegression()

In [18]:
classification.fit(X_train, y_train)


LogisticRegression()

In [19]:
y_pred = classification.predict(X_test)


In [20]:
from sklearn import metrics
print(metrics.classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.86      1.00      0.92      1345
           1       0.00      0.00      0.00       221

    accuracy                           0.86      1566
   macro avg       0.43      0.50      0.46      1566
weighted avg       0.74      0.86      0.79      1566



/opt/anaconda3/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/opt/anaconda3/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/opt/anaconda3/lib/python3.9/site-packages/sklearn/metrics/_classification.py:1248: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


In [21]:
from sklearn.metrics import accuracy_score
accuracy_score(y_test, y_pred)

0.8588761174968071

# With 86% Accuracy you will get to know the is this particular purchase your roduct or not 

## you will check & know who is your targeted Audience & customer

### If it is purchase yopur product then output = 1
### if not then = 0 